src/history.js

```diff
@@ -187,14 +187,14 @@ class History extends Emitter {
     if (this.size <= 0) {
       this.begin()
       return
-    } else if (!next) {
+    } else if (action === this.head) {
       next = this.head = parent
     } else if (action === this.root) {
       this.root = next
     }
 
     if (!action.disabled) {
-      this.reconcile(next)
+      this.reconcile(this.head)
```

Comment:
Not sure if I'm thinking about this too much, but is there a lot of logic that needs to go into this line right here? I''m imagining this like I picture git branches and interactive rebases that remove commits. So for a focused history branch I see removing an action causing a reconcile from that action's parent, and for an unfocused branch i guess i picture no reconciliation, but one would need to occur when you switched from branch to branch starting at the last shared action. Is this can of worms something we need to worry about?